### PR TITLE
feat(jenkins): Automatically set a build description that links back to Spinnaker

### PR DESCRIPTION
**Pull request by ajordens**
Requires that `spinnaker.wwwHost` be set in your `orca` configuration.

```
spinnaker:
  wwwHost: https://www.spinnaker.net
```

image

### CODE REVIEW

**Comment by ajordens** [comment]
Ended up going with `spinnaker.wwwHost` as there's some prior art at Netflix around `spinnaker.apiHost`.

File: orca-igor/src/main/groovy/com/netflix/spinnaker/orca/igor/tasks/MonitorQueuedJenkinsJobTask.groovy [hunk 1]
```diff
@@ -59,4 +64,48 @@ class MonitorQueuedJenkinsJobTask implements OverridableTimeoutRetryableTask {
...
+   * Update the description of a running Jenkins build and provide a deep link back to the particular execution in
+   * Spinnaker.
+   *
+   * Requires that `wwwHost` be specified _and_ that the build is running (vs. queued).
+   */
+  private void updateBuildDescription(StageExecution stageExecution,
+                                      String jenkinsController,
+                                      String jobName,
+                                      Integer buildNumber) {
+    if (wwwHost == null) {
```
Comment (robzienert) [inline orca-igor/src/main/groovy/com/netflix/spinnaker/orca/igor/tasks/MonitorQueuedJenkinsJobTask.groovy:88]: Do we want an `info` log here potentially to inform existing operators that there's a handy feature that could be unlocked with some additional config?

**Review by robzienert — APPROVED** [review]
Looks great as-is, one question.

**Review by jonsie — APPROVED** [review]

File: orca-igor/src/main/groovy/com/netflix/spinnaker/orca/igor/tasks/MonitorQueuedJenkinsJobTask.groovy (hunk 1 above)
Comment (ajordens) [reply orca-igor/src/main/groovy/com/netflix/spinnaker/orca/igor/tasks/MonitorQueuedJenkinsJobTask.groovy:88]: I was going to add something to https://spinnaker.io/setup/ci/jenkins/, a `log.info()` might not hurt as well.

**Comment by luispollo** [comment]
> Ended up going with `spinnaker.wwwHost` as there's some prior art at Netflix around `spinnaker.apiHost`.

FWIW, we use `spinnaker.baseUrl` in Keel. Looks great to me!

**Comment by ajordens** [comment]
@luispollo you've convinced me to change the existing usage, I'll align on something `baseUrl` focused. Thanks!

**Pull request merged**